data/scrapper/erplogin/session.py

```diff
@@ -0,0 +1,112 @@
+import requests
+
+ERP_HOME_URL = 'https://erp.iitkgp.ac.in/IIT_ERP3/'
+SECURITY_QUESTION_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/getSecurityQues.htm'
+LOGIN_URL = 'https://erp.iitkgp.ac.in/SSOAdministration/auth.htm'
+GET_ACAD_TOKEN_URL = 'https://erp.iitkgp.ac.in/Acad/central_breadth_tt.jsp?action=second'
+
+class ERPSession:
+    """
+    An erp session
+
+    Logs into erp after prompting the user to enter details
+
+    Class Attributes::
+    -self.sessionToken #the erp SessionToken or JSESSIONID (as named by erp)
+    -self.ssoToken    #the SSOToken obtained after loggin into erp
+    -self.academicToken #the academic token, useful for getting the data from erp
+
+    Basic Usage::
+
+    >>>from erplogin.session import ERPSession
+    >>>s = ERPSession(roll_no, password)
+    >>>question = s.get_security_question() # Returns the security question from ERP
+    # User feeds the answer to security question.
+    >>>s.LoginERP(answer)
+
+    Obtain the cookies by accesing the corresponding attributes.
+    """
+
+    headers = {
+        'timeout': '20',
+        'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/51.0.2704.79 Chrome/51.0.2704.79 Safari/537.36',
+    }
+
+    def __init__(self, roll_no, password):
+        self.sess = requests.Session()
+        self.roll_no = roll_no
+        self._password = password
+        self.sessionToken = ERPSession.__generate_session_cookie(self)
+        self.question_answer = ''
+        self.academicToken = ''
+        self.ssoToken = ''
+
+
+    def __generate_session_cookie (self):
+        """
+        Automatically called upon declaration
+        Requests for the session id by requesting for erp home page.
+
+        Arguments:
+        > NIL
+
+        Returns 
+        > sessionToken
+        """
+        response_erp = self.sess.get(ERP_HOME_URL)
+        if response_erp.status_code != 200:
+            return None
+
+        return (response_erp.cookies['JSESSIONID'])
```

Comment:
Use `get` method instead.

Reply:
For a second there I thought you meant change the POST HTTP method to GET method for the request. :stuck_out_tongue: